soda/scientific/soda/scientific/anomaly_detection/anomaly_detector.py

```diff
@@ -0,0 +1,263 @@
+import datetime
+from pathlib import Path
+from typing import Any, Dict, List, Mapping, Optional, Tuple
+
+import pandas as pd
+import yaml
+from pydantic import BaseModel, validator
+from soda.common.logs import Logs
+
+from soda.scientific.anomaly_detection.feedback_processor import FeedbackProcessor
+from soda.scientific.anomaly_detection.models.prophet_model import (
+    FreqDetectionResult,
+    ProphetDetector,
+)
+
+
+class UserFeedback(BaseModel):
+    """Validation model for user feedback data dict in payload."""
+
+    isCorrectlyClassified: Optional[bool] = None
+    isAnomaly: Optional[bool] = None
+    reason: Optional[str] = None
+    freeTextReason: Optional[str] = None
+    skipMeasurements: Optional[str] = None
+
+    @validator("skipMeasurements")
+    def check_accepted_values_skip_measurements(cls, v):
+        accepted_values = ["this", "previous", "previousAndThis", None]
+        assert v in accepted_values, f"skip_measurements must be one of {accepted_values}, but '{v}' was provided."
+        return v
+
+
+class SeverityLevelAreas(BaseModel):
+    """Validates severity levels dicts."""
+
+    greaterThanOrEqual: Optional[float] = None
+    lessThanOrEqual: Optional[float] = None
+
+
+class AnomalyDiagnostics(BaseModel):
+    value: Optional[float] = None
+    fail: Optional[SeverityLevelAreas] = None
+    warn: Optional[SeverityLevelAreas] = None
+    anomalyProbability: Optional[float] = None
+    anomalyPredictedValue: Optional[float] = None
+    anomalyErrorSeverity: str = "pass"
+    anomalyErrorCode: str = ""
+    feedback: Optional[UserFeedback] = UserFeedback()
+
+
+class LocationModel(BaseModel):
+    filePath: Optional[str] = None
+    line: Optional[int] = None
+    col: Optional[int] = None
+
+
+# some of those fields might end up being ignored down the line by ADS
+class AnomalyResult(BaseModel):
+    identity: Optional[str] = None
+    measurementId: Optional[str] = None
+    type: Optional[str] = None
+    definition: Optional[str] = None
+    location: LocationModel = LocationModel()
+    metrics: Optional[List[str]] = None
+    dataSource: Optional[str] = None
+    table: Optional[str] = None
+    partition: Optional[str] = None
+    column: Optional[str] = None
+    outcome: Optional[str] = None
+    diagnostics: AnomalyDiagnostics = AnomalyDiagnostics()
+
+
+class AnomalyHistoricalCheckResults(BaseModel):
+    results: List[AnomalyResult]
+
+
+class AnomalyHistoricalMeasurement(BaseModel):
+    id: str
+    identity: str
+    value: float
+    dataTime: datetime.datetime
+
+
+class AnomalyHistoricalMeasurements(BaseModel):
+    results: List[AnomalyHistoricalMeasurement]
+
+
+MOCK_ANOMALY_DIAGNOSTICS = [
```

Comment:
maybe we can remove this part and move it to tests

Reply:
I have something in the tests so I think I'll just remove that, good catch.